diff -crB a/lib/Driver/ToolChain.cpp b/lib/Driver/ToolChain.cpp
*** a/lib/Driver/ToolChain.cpp	2015-12-08 21:21:55.284355136 -0700
--- b/lib/Driver/ToolChain.cpp	2015-12-08 21:24:17.728361457 -0700
***************
*** 22,29 ****
--- 22,32 ----
  #include "llvm/Option/Option.h"
  #include "llvm/Support/ErrorHandling.h"
  #include "llvm/Support/FileSystem.h"
+ #include "llvm/Support/TargetParser.h"
+ 
  using namespace clang::driver;
  using namespace clang;
+ using namespace llvm;
  using namespace llvm::opt;
  
  static llvm::opt::Arg *GetRTTIArgument(const ArgList &Args) {
***************
*** 316,324 ****
      StringRef Suffix = 
        tools::arm::getLLVMArchSuffixForARM(CPU,
                                            tools::arm::getARMArch(MArch, Triple));
!     bool ThumbDefault = Suffix.startswith("v6m") || Suffix.startswith("v7m") ||
!       Suffix.startswith("v7em") ||
!       (Suffix.startswith("v7") && getTriple().isOSBinFormatMachO());
      // FIXME: this is invalid for WindowsCE
      if (getTriple().isOSWindows())
        ThumbDefault = true;
--- 319,327 ----
      StringRef Suffix = 
        tools::arm::getLLVMArchSuffixForARM(CPU,
                                            tools::arm::getARMArch(MArch, Triple));
!     bool IsMProfile = ARM::parseArchProfile(Suffix) == ARM::PK_M;
!     bool ThumbDefault = IsMProfile || (ARM::parseArchVersion(Suffix) == 7 &&
!                                        getTriple().isOSBinFormatMachO());
      // FIXME: this is invalid for WindowsCE
      if (getTriple().isOSWindows())
        ThumbDefault = true;
***************
*** 328,336 ****
      else
        ArchName = "arm";
  
!     // Assembly files should start in ARM mode.
!     if (InputType != types::TY_PP_Asm &&
!         Args.hasFlag(options::OPT_mthumb, options::OPT_mno_thumb, ThumbDefault))
      {
        if (IsBigEndian)
          ArchName = "thumbeb";
--- 331,339 ----
      else
        ArchName = "arm";
  
!     // Assembly files should start in ARM mode, unless arch is M-profile.
!     if (IsMProfile || (InputType != types::TY_PP_Asm &&
!          Args.hasFlag(options::OPT_mthumb, options::OPT_mno_thumb, ThumbDefault)))
      {
        if (IsBigEndian)
          ArchName = "thumbeb";
